tools/automatic_query_fixer/src/main/java/com/google/cloud/bigquery/utils/queryfixer/QueryFixerMain.java

```diff
@@ -1,19 +1,196 @@
+      // In CLI mode, all the instructions are output by print functions, because logger outputs
+      // extract info (time, code position) that distracts users.
+      System.out.println(
+          "Please provide the query as an argument, enclosed by double quote. Use --help for instruction.");
+    }
+
+    String query = cmd.getArgList().get(0);
+    System.out.println("Input query: " + query);
+
+    AutomaticQueryFixer queryFixer = new AutomaticQueryFixer(bigQueryOptions);
+
+    String interactMode = cmd.getOptionValue("interact");
+    if (interactMode == null || interactMode.equalsIgnoreCase("none")) {
+      // todo: Implement Non-interactive mode
+      FixResult fixResult = queryFixer.fix(query);
+      if (fixResult.getOptions().isEmpty()) {
+        return;
+      }
+      String newQuery = fixResult.getOptions().get(0).getFixedQuery();
+      printQueryResult(newQuery, bigQueryOptions);
+
+    } else if (interactMode.equalsIgnoreCase("guide")) {
+      // todo: Implement guide mode
+      return;
+    } else if (interactMode.equalsIgnoreCase("all") || interactMode.equalsIgnoreCase("full")) {
+      FixResult fixResult = fullInteractMode(queryFixer, query);
+      printFixResult(fixResult, cmd.getOptionValue("output"));
+    } else {
+      System.out.println("Interact Mode (-i) is incorrect. Use --help for usage.");
+      System.exit(1);
+    }
+  }
+
+  private static BigQueryOptions buildBigQueryOptions(String credentialPath, String projectId) {
+    if (credentialPath == null) {
+      return BigQueryOptions.newBuilder().setProjectId(projectId).build();
+    } else {
+      // TODO: should support this in near future.
+      System.out.println("customized credential is not supported");
+      System.exit(1);
+      return null;
+    }
+  }
+
+  private static FixResult fullInteractMode(AutomaticQueryFixer queryFixer, String query) {
+    return queryFixer.fix(query);
+  }
+
+  private static void printFixResult(FixResult fixResult, String outputFormat) {
+    if (outputFormat == null || outputFormat.equalsIgnoreCase("natural")) {
+      printFixResultInCommandLine(fixResult);
+    } else if (outputFormat.equalsIgnoreCase("json")) {
+      printFixResultAsJson(fixResult);
+    } else {
+      System.out.println("Output Mode (-o) is incorrect. Use --help for usage.");
+      System.exit(1);
+    }
+  }
+
+  private static void printFixResultAsJson(FixResult fixResult) {
+    Gson gson = new GsonBuilder().setPrettyPrinting().serializeNulls().create();
+    System.out.println(gson.toJson(fixResult));
+  }
+
+  private static void printFixResultInCommandLine(FixResult fixResult) {
+    if (fixResult.getStatus() == FixResult.Status.NO_ERROR) {
+      System.out.println("The input query is correct");
+      return;
+    }
+
+    System.out.println("The query has an error: " + fixResult.getError());
+
+    if (fixResult.getStatus() == FixResult.Status.FAILURE) {
+      System.out.println("The input query is unable to fix");
```

Comment:
"Failed to fix the input query."

Reply:
or "Error occurred when fixing the input query."

Reply:
Change to "Failed to fix the input query."